File: RosettaCode/perl/RSortingBeadsort.pl
#!/usr/bin/env perl
use strict;
use warnings;

sub beadsort {
	my @data = @_;

	my @columns;
	my @rows;

	for my $datum (@data) {
		for my $column ( 0 .. $datum-1 ) {
		++ $rows[ $columns[$column]++ ];
		}
	}

	return reverse @rows;
}

my @result = beadsort 5, 7, 1, 3, 1, 1, 20;
print "@result";
print "\n";

1;

__DATA__
__END__

File: RosettaCode/ruby/RSortingBeadsort.rb
#!/usr/bin/env ruby

class Array
	def beadsort
		self.map {|e| [1] * e}.columns.columns.map {|e| e.length}
	end

	def columns
		y = self.length
		x = self.map {|l| l.length}.max

		Array.new(x) do |row|
			Array.new(y) { |column|
				self[column][row]
			}.compact # Remove nulls.
		end
	end
end
 
# Demonstration code:
harry = [5,3,1,7,4,1,1]
puts harry.inspect
puts harry.beadsort.inspect

